hazelcast/src/main/java/com/hazelcast/cluster/impl/operations/ClusterStartInfoUpdateOperation.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.cluster.impl.operations;
+
+import com.hazelcast.cluster.impl.ClusterServiceImpl;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.util.Clock;
+
+import java.io.IOException;
+
+/**
+ * When a node wants to join the cluster, cluster start time is sent
+ * with joining node and saved in {@link com.hazelcast.cluster.impl.ClusterClockImpl}
+ * of {@link com.hazelcast.cluster.impl.ClusterServiceImpl} This information can
+ * be used to calculate the uptime of the cluster.
+ */
+public class ClusterStartInfoUpdateOperation extends AbstractClusterOperation implements JoinOperation {
+
+    private long startTime = Clock.currentTimeMillis();
```

Comment:
I have no idea what nanoseconds divided by a long number is. Please use `TimeUnit.NANOSECONDS(System.nanoTime().toWhatEverYouNeed()` instead :)